tests/Doctrine/Tests/ORM/Functional/Ticket/GH10449Test.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping as ORM;
+use Doctrine\ORM\Mapping\MappingException;
+use Doctrine\Tests\OrmTestCase;
+
+use function get_parent_class;
+use function method_exists;
+
+class GH10449Test extends OrmTestCase
+{
+    public function testToManyAssociationOnMappedSuperclassShallBeRejected(): void
+    {
+        $em      = $this->getTestEntityManager();
+        $classes = [GH10449MappedSuperclass::class, GH10449Entity::class, GH10449ToManyAssociationTarget::class];
+
+        $this->expectException(MappingException::class);
+        $this->expectExceptionMessageMatches('/illegal to put an inverse side one-to-many or many-to-many association on mapped superclass/');
+
+        foreach ($classes as $class) {
+            $cm = $em->getClassMetadata($class);
+        }
```

Comment:
For which of the classes do we expect the exception? This test will pass if `getClassMetadata()` throws for only one of them, no matter which one.

Reply:
good point, will fix